deals-add.js

```diff
@@ -0,0 +1,223 @@
+var script = document.createElement('script');
```

Comment:
You'll need to wrap it in a document on ready. i.e.
```
$(function () {
<CODE BODY HERE>
});
```
Otherwise, your code will run before your dom elements (i.e. your `document.getElementsByClassName/id` will fail)

Reply:
https://learn.jquery.com/using-jquery-core/document-ready/